integration_test/migration_test.go

```diff
@@ -19,8 +19,41 @@ import (
 	"os"
 	"strings"
 	"testing"
+
+	"sigs.k8s.io/krew/pkg/constants"
 )
 
+func TestKrewIndexAutoMigration(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithEnv(constants.EnableMultiIndexSwitch, 1).WithDefaultIndex()
+	prepareOldIndexLayout(test)
+
+	// any command here should cause the index migration to occur
+	test.Krew("index", "list").RunOrFail()
+	if !isIndexMigrated(test) {
+		t.Error("output should include the default index after migration")
+	}
+}
+
+func TestKrewMigrationSkippedWithNoCommand(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithEnv(constants.EnableMultiIndexSwitch, 1).WithDefaultIndex()
+	prepareOldIndexLayout(test)
+
+	test.Krew().RunOrFail()
+	if isIndexMigrated(test) {
```

Comment:
Is that behaviour something we really require or just a coincidence of the implementation? If the latter is the case, consider removing the test, because it doesn't test a relevant behaviour.

Reply:
I think just a coincidence of the implementation (cobra doesn't seem to execute the prerun in the case of no subcommand or `krew -h`). It was something I noticed when I was testing the migration locally and wasn't really sure if I should add a test for or not. I can remove this if it seems unnecessary.